mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/ExecuteStageRequestService.java

```diff
@@ -36,26 +37,40 @@
 import rx.functions.Func1;
 import rx.subjects.PublishSubject;
 
-
+@Slf4j
 public class ExecuteStageRequestService extends BaseService {
 
     private static final Logger logger = LoggerFactory.getLogger(ExecuteStageRequestService.class);
-    private Observable<WrappedExecuteStageRequest> executeStageRequestObservable;
-    private Observer<Observable<Status>> tasksStatusObserver;
-    private WorkerExecutionOperations executionOperations;
-    private Optional<String> jobProviderClass;
-    private Job mantisJob;
+    private final Observable<WrappedExecuteStageRequest> executeStageRequestObservable;
+    private final Observer<Observable<Status>> tasksStatusObserver;
+    private final WorkerExecutionOperations executionOperations;
+    private final Optional<String> jobProviderClass;
+
+    private final ClassLoaderHandle classLoaderHandle;
+    /** The classpaths used by this task. */
+    private final Collection<URL> requiredClasspaths;
+    private final Job mantisJob;
+
+    /**
+     * This class loader should be set as the context class loader for threads that may dynamically
+     * load user code.
+     */
+    private UserCodeClassLoader userCodeClassLoader;
     private Subscription subscription;
 
-    public ExecuteStageRequestService(Observable<WrappedExecuteStageRequest> executeStageRequestObservable,
-                                      Observer<Observable<Status>> tasksStatusObserver,
-                                      WorkerExecutionOperations executionOperations,
-                                      Optional<String> jobProviderClass,
-                                      Job mantisJob) {
+    public ExecuteStageRequestService(
+        Observable<WrappedExecuteStageRequest> executeStageRequestObservable,
+        Observer<Observable<Status>> tasksStatusObserver,
+        WorkerExecutionOperations executionOperations,
+        Optional<String> jobProviderClass,
+        ClassLoaderHandle classLoaderHandle, Collection<URL> requiredClasspaths,
+        Job mantisJob) {
```

Comment:
nit: make this nullable

Reply:
Job is used in the spring boot path. We need to figure out how to load class for spring boot and get the Job bean.

Reply:
I haven't looked too deep into this but I'm guessing this should be some version of the code you have here: https://sourcegraph.netflix.net/stash.corp.netflix.com/CPIE/nfmantis/-/blob/runtime-spring-boot/src/main/java/com/netflix/mantis/runtime/command/SpringReadJobFromZip.java?L65#tab=references

Basically, create an AnnotationBasedApplicationContext based on the classloader and then get the bean for the job class. I'll try to get a test spring-boot app working after the nodequark job migration.